plugins/imfile/imfile.c

```diff
@@ -2522,6 +2562,204 @@ do_inotify(void)
+	DBGPRINTF("imfile: fen_processEvent: associate port for %s\n", fobjp->fo_name);
+	if (port_associate(port, PORT_SOURCE_FILE, (uintptr_t)fobjp,
+				pLstn->pfinf->events, (void *)pLstn) == -1) {
+		/* Add error processing as required, file may have been deleted/moved. */
+		errmsg.LogError(1, RS_RET_SYS_ERR, "imfile: fen_processEvent: Failed to register file "
+			": %s - errno %d\n", fobjp->fo_name, errno);
+		free(pLstn->pfinf->fobj.fo_name);
+		free(pLstn->pfinf);
+		pLstn->pfinf = NULL;
+		ABORT_FINALIZE(RS_RET_SYS_ERR);
+	} else {
+		/* Port successfull listening now*/
+		pLstn->bPortAssociated = 1;
+	}
+
+finalize_it:
+	RETiRet;
+}
+
+static rsRetVal
+do_fen(void)
+{
+	int port;
+	int bHadFileData; /* were there at least one file with data during this run? */
+	port_event_t portEvent;
+	struct timespec timeout;
+	lstn_t *pLstn;	/* Listener helper*/
+	DEFiRet;
+
+	/* Set port timeout to 1 second. We need to checkfor unmonitored files during meantime */
+	timeout.tv_sec = 1;
+	timeout.tv_nsec = 0;
+
+	/* create port instance */
+	if ((port = port_create()) == -1) {
+		errmsg.LogError(1, RS_RET_INOTIFY_INIT_FAILED, "imfile: do_fen: Init port failed ");
+		return RS_RET_INOTIFY_INIT_FAILED;
+	}
+
+	/* Loop through all configured listeners */
+	for(pLstn = runModConf->pRootLstn ; pLstn != NULL ; pLstn = pLstn->next) {
+		if(pLstn->masterLstn == NULL) {
+			// Create FileInfo struct
+			pLstn->pfinf = malloc(sizeof(struct fileinfo));
+			if (pLstn->pfinf == NULL) {
+				errmsg.LogError(1, RS_RET_INOTIFY_INIT_FAILED, "imfile: do_fen: alloc memory "
+					"for fileinfo failed ");
+				ABORT_FINALIZE(RS_RET_INOTIFY_INIT_FAILED);
+			}
+
+			if ((pLstn->pfinf->fobj.fo_name = strdup((char*)pLstn->pszFileName)) == NULL) {
+				errmsg.LogError(1, RS_RET_INOTIFY_INIT_FAILED, "imfile: do_fen: alloc memory "
+					"for strdup failed ");
+				free(pLstn->pfinf);
+				pLstn->pfinf = NULL;
+				ABORT_FINALIZE(RS_RET_INOTIFY_INIT_FAILED);
+			}
+
+			/* Event types to watch. */
+			pLstn->pfinf->events = FILE_MODIFIED; // |FILE_DELETE|FILE_RENAME_TO|FILE_RENAME_FROM;
+			pLstn->pfinf->port = port;
+
+			/* Call file events event handler */
+			fen_processEvent(pLstn, pLstn->pfinf->events);
+		}
+	}
+
+	DBGPRINTF("imfile: do_fen: ENTER monitoring loop \n");
+	while(glbl.GetGlobalInputTermState() == 0) {
+		DBGPRINTF("imfile: do_fen: loop begin... \n");
+		/* Loop through events, if there are any */
+		while (!port_get(port, &portEvent, &timeout)) {
+			/*
+			* Can add cases for other sources if this
+			* port is used to collect events from multiple sources.
+			*/
+			DBGPRINTF("imfile: do_fen: event received, processing ... \n");
+			switch (portEvent.portev_source) {
+				case PORT_SOURCE_FILE:
+					/* Call file events event handler */
+					fen_processEvent((lstn_t*)portEvent.portev_user, portEvent.portev_events);
+					break;
+				default:
+					errmsg.LogError(1, RS_RET_SYS_ERR, "imfile: do_fen: Event from unexpected source "
```

Comment:
```LogError()``` is now available natively (without ```errmsg.``` in front of it). I suggest to switch to that variant (it's even slightly faster, what really doesn't matter here ;-))